openshift/helper/__init__.py

```diff
@@ -127,8 +439,220 @@ def __lookup_method(self, operation, namespaced):
         return method
 
     @staticmethod
-    def __get_model(api_version, kind):
-        camel_kind = string_utils.snake_case_to_camel(kind).capitalize()
-        model_name = api_version.capitalize() + camel_kind
-        model = getattr(client.models, model_name)
+    def get_model(api_version, kind):
+        """
+        Return the model class for the requested object.
+
+        :param api_version: API version string
+        :param kind: The name of object type (i.e. Service, Route, Container, etc.)
+        :return: class
+        """
+        camel_kind = string_utils.snake_case_to_camel(kind)
+        # capitalize the first letter of the string without lower-casing the remainder
+        name = camel_kind[:1].capitalize() + camel_kind[1:]
+        model_name = api_version.capitalize() + name
+        try:
+            model = getattr(client.models, model_name)
+        except Exception as exc:
+            raise OpenShiftException(
+                    "Error: openshift.client.models.{} was not found. "
+                    "Did you specify the correct Kind and API Version?".format(model_name)
+            )
         return model
+
+    @property
+    def argspec(self):
```

Comment:
I'm not inclined to move this into module_utils, or have module utils essentially copy the argspec property. It does make module_utils reliant on a magical black box but the only way to really prevent that, I think, is to surface openshift.client within module_utils.